File: go/two-fer/two_fer.go
package twofer

import (
	"fmt"
)

// ShareWith returns a sentence given a name or lack thereof.
func ShareWith(s string) string {
	var noun string
	if len(s) == 0 {
		noun = "you"
	} else {
		noun = s
	}
	return fmt.Sprintf("One for %v, one for me.", noun)
}

File: cpp/two-fer/two_fer.cpp
#include "two_fer.h"

namespace two_fer {

    const std::string two_fer() {
        return "One for you, one for me.";
    }

    const std::string two_fer(const std::string &name) {
        return "One for " + name + ", one for me.";
    }

} // namespace two_fer
